src/device.rs

```diff
@@ -2312,6 +2308,21 @@ impl Runtime {
             }
         }
     }
+
+    fn is_dublicated_event(&mut self, node: &Node) -> bool {
```

Comment:
```suggestion
    fn is_duplicated_event(&mut self, node: &Node) -> bool {
```

Reply:
Why can't this function be just `*node == *last_node` check? I fear a bit that the implicit `state` checking hides a possible bug since if there was no previous event it means the passed event is guaranteed not to be a duplicate.

Reply:
One could do the `node == last_node` check, but that would require hashmap `last_transmitted_event` entries to never be removed. This is obviously undesirable, therefore whenever `Disconnect` event is fired - the `last_transmitted_event` entry is cleared from hashmap. 

This slightly changes the semantics, because event missing from hasmap now can mean one of two things:
* Either there was never an event fired for specified `public_key`
* Or the last event fired for specified `public_key` is Disconnected.

Because `Disconnected` event can never be first event fired for the node -> we can safely assume, that if event has state `disconnected` and there is no entry in hashmap -> the entry is dublicated `Disconnect` event.

Reply:
Ok that makes sense

Reply:
It should still be renamed `dublicated` -> `duplicated`, but I'm also okay with that happening in the follow-up